src/classes/FeatureParameterMapper_TEST.cls

```diff
@@ -66,18 +66,174 @@ private class FeatureParameterMapper_TEST {
+             FeatureParameterMapper.TelemetryParameterName convertedValue = FeatureParameterMapper.getInstance().convertToEnum('Test');
+        Test.stopTest();
+
+        System.assertEquals(null, convertedValue, 'When there is no match null should be returned');
+    }
+
     /**************************************************************************************************************************
     * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
     ***************************************************************************************************************************/
     @isTest
-    private static void getFeatureParameterModel() {
+    private static void getFeatureParameterModelStringNotAnEnum() {
+
+        try {
+            Test.startTest();
+                FeatureParameterModel featureParamModel = FeatureParameterMapper.getInstance().getFeatureParameterModel('testFeatureParam');
+                System.assert(false, 'When Passed parameter string is not an enum an error is thrown');
+            Test.stopTest();
+        } catch (FeatureParameterMapper.FeatureParameterMapperException ex) {
+            System.assert(ex.getMessage().contains('An error in Feature Parameter'));
+        }
+
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterModelEnumNotDate() {
+        try {
+            Test.startTest();
+                FeatureParameterModel featureParamModel = FeatureParameterMapper.getInstance().getFeatureParameterModel('HasUserManagedTDTM');
+                System.assert(false, 'When Passed parameter string is not an enum an error is thrown');
+            Test.stopTest();
+        } catch (FeatureParameterMapper.FeatureParameterMapperException ex) {
+            System.assert(ex.getMessage().contains('An error in Feature Parameter'));
+        }
+
+    }
+
+    /**************************************************************************************************************************
+    * @description Test method to verify that the getFeatureParameterModel method returns a FeatureParameterModel
+    ***************************************************************************************************************************/
+    @isTest
+    private static void getFeatureParameterModelSuccess() {
+
+        Date featureParameterDate = Date.valueOf('2015-1-1');
+        String featureParameterName = 'SettingsHealthCheckLastRunDate';
+
+        // Set value for the Feature Parameter
+        System.FeatureManagement.setPackageDateValue(featureParameterName, featureParameterDate);
```

Comment:
@MatthewBlanski wanted to check if we can set the featureParameter value here in the test method or do we want call the update updateFromFeatureParameterModel using stub and then do a getFeatureParameterModel()?